py/torch_tensorrt/dynamo/conversion/_conversion.py

```diff
@@ -70,18 +70,22 @@ def insert_engine_to_cache(
             f"Detected that the engine with hash: {hash_val} exists in cache. It will be refreshed"
         )
 
-    # set EXCLUDE_WEIGHTS flag to strip weights
-    serialization_config = interpreter_result.engine.create_serialization_config()
-    serialization_config.set_flag(trt.SerializationFlag.EXCLUDE_WEIGHTS)
-    weight_stripped_serialized_engine = interpreter_result.engine.serialize_with_config(
-        serialization_config
-    )
+    # for TensorRT >= 10.14, we save weight-stripped engine in cache
+    if hasattr(trt.SerializationFlag, "INCLUDE_REFIT"):
+        # set EXCLUDE_WEIGHTS flag to strip weights
+        serialization_config = interpreter_result.engine.create_serialization_config()
+        serialization_config.set_flag(trt.SerializationFlag.EXCLUDE_WEIGHTS)
+        serialized_engine_in_cache = interpreter_result.engine.serialize_with_config(
```

Comment:
Not a huge fan of taking advantage of the lack of scope conditions in python. 

Maybe something like this would be nicer 


```py

serialization_config = interpreter_result.engine.create_seralization_config()

if hasattr(trt.SeralizationFlag, "INCLUDE_REFIT"):
    serialization_config.set_flag(trt.SerializationFlag.EXCLUDE_WEIGHTS)
else:
    serialization_config.clear_flag(trt.SerializationFlag.EXCLUDE_WEIGHTS)

serialized_engine = interpreter_result.engine.serialize_with_config(serialization_config)

...
```